src/main/java/uk/ac/cam/cl/dtg/segue/dao/content/GitContentManager.java

```diff
@@ -248,20 +263,92 @@ public final Content getContentDOById(final String version, final String id, boo
+            if (documentTypes == null || documentTypes.contains(documentType)) {
+                BooleanMatchInstruction contentQuery = new BooleanMatchInstruction();
+
+                // Match document type
+                contentQuery.must(new MustMatchInstruction(Constants.TYPE_FIELDNAME, documentType));
+
+                // Generic pages must be explicitly tagged to appear in search results
+                if (documentType.equals(PAGE_TYPE)) {
+                    contentQuery.must(new MustMatchInstruction(Constants.TAGS_FIELDNAME, SEARCHABLE_TAG));
+                }
+
+                // Try to match fields
+                for (String field : importantFields) {
+                    contentQuery.should(new ShouldMatchInstruction(field, searchString, 10L, false));
+                    contentQuery.should(new ShouldMatchInstruction(field, searchString, 5L, true));
+                }
+                for (String field : otherFields) {
+                    contentQuery.should(new ShouldMatchInstruction(field, searchString, 5L, false));
+                    contentQuery.should(new ShouldMatchInstruction(field, searchString, 1L, true));
+                }
+
+                // Check location.address fields on event pages
+                if (documentType.equals(EVENT_TYPE)) {
+                    String addressPath = String.join(nestedFieldConnector, Constants.ADDRESS_PATH_FIELDNAME);
+                    for (String addressField : Constants.ADDRESS_FIELDNAMES) {
+                        String field = addressPath + nestedFieldConnector + addressField;
+                        contentQuery.should(new ShouldMatchInstruction(field, searchString, 3L, false));
+                        contentQuery.should(new ShouldMatchInstruction(field, searchString, 1L, true));
+                    }
+                }
+
+                if (documentType.equals(EVENT_TYPE)){
+                    LocalDate today = LocalDate.now();
+                    long now = today.atStartOfDay(ZoneId.systemDefault()).toEpochSecond() * Constants.EVENT_DATE_EPOCH_MULTIPLIER;
+
+                    // Strongly prefer future events
+                    contentQuery.should(new RangeMatchInstruction<Long>(Constants.DATE_FIELDNAME).greaterThanOrEqual(now).boost(20));
+                    contentQuery.should(new RangeMatchInstruction<Long>(Constants.DATE_FIELDNAME).lessThan(now).boost(1));
+
+                    // Every event will match one of these "should" queries so we increase the minimum expected matches
+                    numberOfExpectedShouldMatches += 1;
+                }
+
+                contentQuery.setMinimumShouldMatch(numberOfExpectedShouldMatches);
+                matchQuery.should(contentQuery);
+            }
+        }
+
+        // Do not include any content with a nofilter tag
+        matchQuery.mustNot(new MustMatchInstruction(Constants.TAGS_FIELDNAME, HIDE_FROM_FILTER_TAG));
```

Comment:
We'd sort of like staff to be able to find these using search, but getting the user information down to this level would be a pain. Perhaps an overriden method with another boolean flag for `includeHiddenContent`? I don't know how painful that would be.

Reply:
OK, that sounds easy enough to achieve